### ISSUE TITLE
Choose a random color if none is chosen on collection creation

### ISSUE THREAD

**Issue by tommoor** [outline/outline#3900]
If a new collection is created without a color chosen, select one from the palette automatically. Ideally this logic would live on the server side to account for collections created via API

**Comment by apoorv-mishra** [comment]
I can pick this up if you tell me the default color to use :smile: 

Currently the default one set on FE is `#4E5C6E`. I'm assuming we want to set the same value on BE too.

On another note, does this require a data migration for all the entries where the `color` is `null`?

**Comment by tommoor** [comment]
I'd suggest:

- Change the frontend so that if no color is actively chosen then no color is sent to the backend
- If no color is given to `collections.create` endpoint then choose a random one from the palette
- This may require moving the palette to the `/shared` code so it's accessible from FE and BE
- No migration needed, all existing collections should remain the same

**Comment by apoorv-mishra** [comment]
I uncovered some problems while attempting to achieve the following

>Change the frontend so that if no color is actively chosen then no color is sent to the backend

Supposedly, changing [this](https://github.com/outline/outline/blob/294521f1624fdc4617150d4fa128146e7057013b/app/scenes/CollectionNew.tsx#L33) to

```js
color = "";
```
should lead us to the _no color chosen_ scenario. However, doing so still sets the value for the input field as `000000` which the backend treats as invalid because of [this assertion](https://github.com/outline/outline/blob/294521f1624fdc4617150d4fa128146e7057013b/server/routes/api/collections.ts#L64).

Next, I attempted to set

```js
color = "#";
```
and, it actually set the color to `#000000`.

I also attempted to keep just

```js
color: string;
```
and, for some reason it set the default to some _magic value_, i.e, [`#22194D`](https://github.com/casesandberg/react-color/issues/837). Some further investigation revealed that it's set using [one hex prop](https://github.com/casesandberg/react-color/blob/bc9a0e1dc5d11b06c511a8e02a95bd85c7129f4b/src/components/twitter/Twitter.js#L10) [here](https://github.com/casesandberg/react-color/blob/bc9a0e1dc5d11b06c511a8e02a95bd85c7129f4b/src/components/twitter/Twitter.js#L146). Exactly how? Not sure(there are some [issues](https://github.com/casesandberg/react-color/issues?q=22194D) as well regarding the same).

In short, the `ColorPicker` doesn't seem to allow sending an empty value for the input field yielding a _no color chosen_ scenario. That being said, I wanted to inquire as to why do we want to touch the FE in this case? What's wrong with keeping the default value on FE as it is? Is there a specific use case to be covered? If not, we can just modify the backend to handle it.

**Comment by tommoor** [comment]
> Or, we can do a random color generation on FE as well, similar to BE, and set the value accordingly.

Thanks for the research. Perhaps we could just pick a random color when the create collection modal is mounted then… this would have the benefit of showing the color to the user too.
